tests/test_insert.py

```diff
@@ -195,3 +198,44 @@ def test_single_insert_obs_not_unique(test_session):
     ]
     dbm = single_insert_strategy(test_session, ob)
     assert dbm.skips == 1
+
+
+def test_mult_networks(test_session, caplog):
+    caplog.set_level(logging.DEBUG, "crmprtd")
+
+    moti = "Brandywine"
+    wmb = "FIVE MILE"
+    ec = "Sechelt"
+    stations = [moti, wmb, ec]
+
+    obs = []
+    with test_session:
+        for station in stations:
+            hist = (
+                test_session.query(History)
+                .filter(History.station_name == station)
+                .first()
+            )
+            var = hist.station.network.variables[0]
+            observation = Obs(
+                history_id=hist.id,
+                datum=10,
+                vars_id=var.id,
+                time=datetime(2012, 9, 24, 6, tzinfo=pytz.utc),
+            )
+            obs.append(observation)
+
+    results = insert(test_session, obs)
+    assert networks_logged(obs, test_session, caplog)
+
+
+def networks_logged(obs, test_session, caplog):
+    obs_by_network_dict = obs_by_network(obs, test_session)
```

Comment:
I understand the motivation to reuse code here in the testing. You have a nice function that you've written to group the observations by network, so why not use it for the test? One reason *not* to, is that it essentially results in circular logic. You have a test that executes `obs_by_network()` and you compare that against results that *also* execute `obs_by_network()` :smile: It's like the reflexive property. Anything is always equal to itself.

I'd advise that you write a separate test *specifically* testing `obs_by_network()` under a few different scenarios to ensure that it's working in isolation.

Then `test_multi_networks` might become simpler, or at least you can trust that `obs_by_network()` is safe and your test of `insert()` should be checking other things.